src/cross.js

```diff
@@ -132,6 +139,20 @@ const asResultAccount = (unsafe: mixed): AccountData => {
   if (typeof balance !== "string") {
     throw new Error("invalid account.balance");
   }
+
+  if (swapHistory && Array.isArray(swapHistory)) {
+    for (let s of swapHistory) {
+      if (s && typeof s === "object") {
+        for (let k in s) {
+          const v = s[k];
+          if (v && typeof v !== "string") {
+            throw new Error(`invalid account.swapHistory.${k}`);
```

Comment:
Instead of throwing I would prefer we just ignore swapHistory to not prevent us to change the format in future.

Reply:
At first invalid, we now set a flag and default to `[]`